src/Orleans/CodeGeneration/GrainInterfaceUtils.cs

```diff
@@ -126,6 +126,9 @@ public static bool IsStatelessWorker(MethodInfo methodInfo)
         
         public static int ComputeMethodId(MethodInfo methodInfo)
         {
+            var attr = methodInfo.GetCustomAttribute<MethodIdAttribute>(true);
+            if (attr != null) return attr.MethodId;
```

Comment:
Sorry I'm late here... Just curious... It will fail compilation only while using build-time codegen... What about runtime codegen? Will we need to rely analyzers?